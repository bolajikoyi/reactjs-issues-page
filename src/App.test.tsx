import React from 'react';
import { render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import App from './App';

jest.mock('./components/main/main', () => () => <div>Main component mock</div>)

describe('App component', ()=> {
  test('Expect componet to be created', () => {
    expect(App).toBeTruthy()
  });

  test('renders without crashing', ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
  });

  test('renders main component', ()=> {
    const {container} = render(<App/>);
    expect(container.textContent).toMatch('Main component mock');
  })

  test('match snapshot', ()=> {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})



