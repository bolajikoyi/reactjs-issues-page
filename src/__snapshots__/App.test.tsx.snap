// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component match snapshot 1`] = `
<div
  className="App"
>
  <div>
    Main component mock
  </div>
</div>
`;
